sessionize/app/src/main/java/co/touchlab/sessionize/NotificationsApiImpl.kt

```diff
@@ -0,0 +1,108 @@
+                .setSmallIcon(R.drawable.baseline_insert_invitation_24)
+                .setContentTitle(title)
+                .setContentText(message)
+                .setWhen(timeInMS)
+                .setPriority(NotificationCompat.PRIORITY_DEFAULT)
+                .setCategory(NotificationCompat.CATEGORY_REMINDER)
+
+
+        // Building Intent wrapper
+        val intent = Intent().also { intent ->
+            intent.action = AndroidAppContext.app.getString(R.string.notification_action)
+            intent.putExtra(NotificationPublisher.NOTIFICATION_ID, notificationId)
+            intent.putExtra(NotificationPublisher.NOTIFICATION, builder.build())
+            val componentName = ComponentName(AndroidAppContext.app, NotificationPublisher::class.java)
+            intent.component = componentName
+        }
+
+        val pendingIntent = PendingIntent.getBroadcast(AndroidAppContext.app, notificationId, intent, PendingIntent.FLAG_CANCEL_CURRENT)
+
+        // Scheduling Intent
+        val alarmManager = AndroidAppContext.app.getSystemService(Context.ALARM_SERVICE) as AlarmManager
+        alarmManager.set(AlarmManager.RTC_WAKEUP, timeInMS, pendingIntent)
+    }
+
+    override fun cancelLocalNotification(notificationId: Int) {
+        with(NotificationManagerCompat.from(AndroidAppContext.app)) {
+            this.cancel(notificationId)
+        }
+    }
+
+    override fun initializeNotifications() {
+        val filter = IntentFilter(AndroidAppContext.app.getString(R.string.notification_action))
+        AndroidAppContext.app.registerReceiver(notificationPublisher, filter)
+
+        createNotificationChannel()
+    }
+
+    override fun deinitializeNotifications() {
+    }
+
+    class NotificationPublisher : BroadcastReceiver() {
+
+        override fun onReceive(context: Context, intent: Intent) {
+            val notification = intent.getParcelableExtra<Notification>(NOTIFICATION)
+            val notificationId = intent.getIntExtra(NOTIFICATION_ID, 0)
+
+            with(NotificationManagerCompat.from(AndroidAppContext.app)) {
+                // notificationId is a unique int for each notification that you must define
+                this.notify(notificationId, notification)
+            }
+        }
+
+        companion object {
+            var NOTIFICATION_ID = "notification_id"
+            var NOTIFICATION = "notification"
+        }
+    }
+
+    private fun createNotificationChannel() {
+        // Create the NotificationChannel, but only on API 26+ because
+        // the NotificationChannel class is new and not in the support library
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val name = AndroidAppContext.app.getString(R.string.notification_channel_name)
+            val descriptionText = AndroidAppContext.app.getString(R.string.notification_channel_description)
+            val channelId = AndroidAppContext.app.getString(R.string.notification_channel_id)
```

Comment:
It seems strange to put these in strings.xml considering it takes the extra effort to get them and they are specifically not translatable. Perhaps a private constant in a companion object would work?